src/transformers/models/marian/modeling_marian.py

```diff
@@ -1083,7 +1090,11 @@ def __init__(self, config: MarianConfig):
         super().__init__(config)
 
         padding_idx, vocab_size = config.pad_token_id, config.vocab_size
-        self.shared = nn.Embedding(vocab_size, config.d_model, padding_idx)
+
+        if config.share_encoder_decoder_embeddings:
```

Comment:
As discussed offline, I think we should rewrite this a bit. Even if `config.share_encoder_decoder_embeddings is False` we should clone those weights to initialized encoder and decoder with it. This makes this PR much more appealing for existing Marian models. E.g. one could then simple load a Marian model with:


```python
from transformers import MarianMTModel

model = MarianMTModel.from_pretrained("Helsinki-NLP/opus-mt-en-de", share_encoder_decoder_embeddings=False)

# now one can fine-tune the model with encoder / decoder embeddings not being shared which has been shown to lead to improved results
```